sql/hive/src/test/scala/org/apache/spark/sql/sources/SimpleTextHadoopFsRelationSuite.scala

```diff
@@ -69,18 +69,19 @@ class SimpleTextHadoopFsRelationSuite extends HadoopFsRelationTest with Predicat
   test("test hadoop conf option propagation") {
     withTempPath { file =>
       // Test write side
+      val nullVal: String = null
```

Comment:
I have hesitated to submit this PR due to this problem for few weeks. After thinking hard, I decided to submit this.

The test below:

```scala
...
 .option("some-null-value-option", null) 
```

was testing of setting `null`. Because `String` is the only reference type (we have `Double`, `Long` and `Boolean` which are `AnyVal` and they can't be null), 
it was fine so far.

Now, `Array[String]` is added and the compiler gose confused and therefore we should explictly give the type to `null`.
Unless we do a runtime checking, it seems we can't add overrien versions anymore without fixing this test.

So, yes, it will break the APIs if any users were setting `null` without type but I wonder if it is worth to keep this behaviour. There are a lot of APIs that do not allow `null` without type so far, e.g., `functions.array(null)`.